TST: use sympy pre-releases in bleeding-edge workflow

neutrinoceros opened the pull request:
Looks like sympy 1.10rc2 breaks unyt, see https://github.com/sympy/sympy/issues/23148

I propose we start using sympy pre-releases in the daily "bleeding-edge" workflow so these problems are always caught in a timely fashion.

oscarbenjamin commented:
I intend to make more frequent pre-releases of sympy in future so this would be useful to pick up on issues long before the main release.

neutrinoceros commented:
Looks like normal CI is broken too, for other reasons, I don't know if I'll be able to investigate this today.

neutrinoceros commented:
So one of the problems was that black made some style changes since the last time CI was run successfully. This problem and a possible solutions are discussed in #193. Pinning black for now.
Another problem is that the last version of pytest-doctestplus that is available for Python 3.6 is broken by the latest version of pytest
https://github.com/astropy/pytest-doctestplus/issues/170

I'll pin that too.

neutrinoceros commented:
I didn't manage to constrain pytest<7 just for Python 3.6 so I'm pinning it globally in tox.ini

neutrinoceros commented:
Ran the workflow on my fork again with sympy 1.10rc3 and it passed https://github.com/neutrinoceros/unyt/runs/5345360352?check_suite_focus=true

chrishavlin commented on .github/workflows/ci.yml:
```diff
@@ -5,6 +5,8 @@ on:
     - master
   pull_request:
+  schedule:
+    - cron:  '1 0 * * *'
```
What's the benefit to running the standard ci every day in addition to the beleeding-edge?

neutrinoceros replied:
The bleeding edge workflow didn't detect the formatting changes from black, so it's clearly not covering everything that normal CI does. My reasoning is that it's easier to just run both daily than trying to cover everything with bleeding edge.

chrishavlin reviewed (approved):
Thanks for figuring this out!

neutrinoceros commented:
To @oscarbenjamin's request (https://github.com/sympy/sympy/issues/22241#issuecomment-1061302909), I'm adding a test for unpicking unyt arrays with different versions of sympy. Hopefully this should help remove some of the maintenance burden from sympy.

oscarbenjamin commented:
> I'm adding a test for unpicking unyt arrays with different versions of sympy.

The best thing would be to add a test along these lines to sympy itself rather than unyt. I don't know what's in these pickles but can you make a representative example that doesn't actually require unyt to be installed?

Ideally I'd want a test that's just like
```python
pickle = b'...'
assert loads(pickle) == ...
```
Then we could add pickles for the same expression generated with all previous versions of sympy and see that they all unpickle correctly on current sympy master.

neutrinoceros commented:
I may give it a try in a few days. ATM I'm not sure how to minimise the test case without missing the point as we did 6 month ago.

neutrinoceros commented:
@jzuhone could you please have a look ? CI is still very unstable on unyt without these patches
